python/dbsp/config.py

```diff
@@ -17,92 +18,169 @@
+            workers=workers,
+            inputs=PipelineConfigInputs(),
+            outputs=PipelineConfigOutputs()
         )
         self.config_id = None
         self.config_version = None
-        # self.workers = workers
-        # print("config: " + str(self.pipeline_config))
+        self.attached_connectors = []
+        self.pipeline_id = None
+        self.name = name
+        self.description = description
 
-    def add_input(self, name: str, input_endpoint_config: InputEndpointConfig):
+    def add_input(self, stream: str, input_endpoint_config: DBSPConnector):
         """Add an input endpoint to the pipeline configuration.
 
         Args:
-            name (str): Endpoint name (must be unique across input endpoints).
+            stream (str): Input name (what table to connect to).
             input_endpoint_config (InputEndpointConfig): Endpoint configuration.
         """
+        input_endpoint_config.save()
 
-        # print("yaml:\n" + str(yaml.dump(input_endpoint_config.to_dict())))
-        self.pipeline_config.inputs[name] = input_endpoint_config
+        self.attached_connectors.append(AttachedConnector(
+            connector_id=input_endpoint_config.connector_id,
+            direction=Direction.INPUT,
+            uuid=uuid.uuid4().hex,
+            config=stream,
+        ))
 
-    def add_file_input(self, stream: str, filepath: str, format_: FormatConfig, **file_config_options):
+    def add_kafka_input(self, name: str, stream: str, config: KafkaInputConfig, format: FormatConfig):
         """Add an input endpoint that reads data from a file to the pipeline configuration.
 
         Args:
-            stream (str): Input stream name to connect the endpoint to.
+            name (str): Name of the input endpoint.
+            stream (str): The input table to connect the endpoint to.
             filepath (str): File to read data from.
             format_ (FormatConfig): Data format specification, e.g., CsvInputFormatConfig().
             file_config_options: Additional configuration options defined in FileInputConfig.
         """
         self.add_input(
-                stream,
-                InputEndpointConfig(
-                    stream = stream,
-                    transport = TransportConfig(
-                        name = "file",
-                        config = FileInputConfig.from_dict(dict({ 'path': filepath }, **file_config_options))),
-                    format_ = format_))
-
-    def add_output(self, name: str, output_endpoint_config: OutputEndpointConfig):
+            stream,
+            DBSPConnector(self.api_client, name, ConnectorType.KAFKAIN, TransportConfig(
+                name="kafka",
+                config=config), format=format))
+
+    def add_kafka_output(self, name: str, stream: str, config: KafkaOutputConfig, format: FormatConfig):
+        """Add an input endpoint that reads data from a file to the pipeline configuration.
+
+        Args:
```

Comment:
This comment no longer matches the function. Is there a better way to document arguments in Python?